freva-data-portal-worker/src/data_portal_worker/backends/posix_and_cloud.py

```diff
@@ -41,18 +42,36 @@ def get_xr_engine(file_path: str) -> Optional[str]:
     except Exception:
         pass
 
+    try:
+        with xr.open_dataset(
+            file_path,
+            engine="h5netcdf",
+            decode_cf=False,
+            use_cftime=False,
+            chunks=None,
+            cache=False,
+            decode_coords=False,
+        ) as _:
+            return "h5netcdf"
+    except Exception:
+        pass
+
     return None
 
 
-def load_posix(inp_file: Union[str, Path]) -> xr.Dataset:
+def posix_and_cloud(inp_file: Union[str, Path]) -> xr.Dataset:
     """Open a dataset with xarray."""
-    inp_file = Path(inp_file)
+    inp_str = str(inp_file)
+    parsed = urlparse(inp_str)
+    target: Union[str, Path]
+    target = Path(inp_str) if parsed.scheme in ("", "file") else inp_str
+    engine = get_xr_engine(str(target))
     return xr.open_dataset(
-        inp_file,
+        target,
         decode_cf=False,
         use_cftime=False,
-        chunks="auto",
+        chunks="auto" if engine != "h5netcdf" else None,
```

Comment:
why is that?

Reply:
Yes that is a super bad idea. Each file will be loaded into memory when openend. We won't do that! If it is not working for netCDF on s3 you have to check whether you have that case and then do something about that. Possibly refuse to do any operation (check with @mannreis  what are options, but I guess not many.)